light/odr_util.go

```diff
@@ -127,15 +126,16 @@ func GetBlock(ctx context.Context, odr OdrBackend, hash common.Hash, number uint
 // GetBlockReceipts retrieves the receipts generated by the transactions included
 // in a block given by its hash.
 func GetBlockReceipts(ctx context.Context, odr OdrBackend, hash common.Hash, number uint64) (types.Receipts, error) {
-	// Retrieve the potentially incomplete receipts from disk or network
-	receipts := rawdb.ReadReceipts(odr.Database(), hash, number)
+	// Assume receipts are already stored locally and attempt to retrieve.
+	receipts := rawdb.ReadRawReceipts(odr.Database(), hash, number)
 	if receipts == nil {
 		r := &ReceiptsRequest{Hash: hash, Number: number}
 		if err := odr.Retrieve(ctx, r); err != nil {
 			return nil, err
 		}
 		receipts = r.Receipts
 	}
+
 	// If the receipts are incomplete, fill the derived fields
 	if len(receipts) > 0 && receipts[0].TxHash == (common.Hash{}) {
```

Comment:
For v4 receipt, the txHash is assigned while some other non-consensus fields are still missing.

Reply:
@rjl493456442  See my response to your comment= in [`accessors_chain.go`](https://github.com/ethereum/go-ethereum/pull/19345#discussion_r274487370). I will hand this similarly based on what we decide.